srcpkgs/blender/patches/0001-musl-fixes.patch

```diff
@@ -14,11 +14,65 @@ Subject: [PATCH 1/2] musl fixes
 +++ b/extern/glog/src/config_linux.h
 @@ -14,7 +14,9 @@
  #define HAVE_DLFCN_H
- 
+
  /* Define to 1 if you have the <execinfo.h> header file. */
 +#ifdef __GLIBC__
  #define HAVE_EXECINFO_H
 +#endif
- 
+
  /* Define if you have the `fcntl' function */
  #define HAVE_FCNTL
+--- a/source/blender/blenlib/BLI_index_range.hh
++++ b/source/blender/blenlib/BLI_index_range.hh
+@@ -38,6 +38,7 @@
+  */
+
+ #include <algorithm>
++#include <cstdint>
+ #include <iosfwd>
+
+ #include "BLI_assert.h"
+
+--- a/source/blender/blenkernel/BKE_volume_enums.hh
++++ b/source/blender/blenkernel/BKE_volume_enums.hh
+@@ -8,7 +8,11 @@
+  * \ingroup bli
+  */
+
+-enum VolumeGridType : int8_t {
++enum VolumeGridType
++#ifdef __GLIBC__
++: int8_t
++#endif
++{
+   VOLUME_GRID_UNKNOWN = 0,
+   VOLUME_GRID_BOOLEAN,
+   VOLUME_GRID_FLOAT,
+
+--- a/extern/lzma/Threads.h
++++ b/extern/lzma/Threads.h
+@@ -82,7 +82,12 @@
+
+ #ifdef Z7_AFFINITY_SUPPORTED
+
++#ifndef __GLIBC__
++typedef struct CCpuSet { unsigned long __bits[128/sizeof(long)]; } CCpuSet;
++// because including sched.h doesn't work
++#else
+ typedef cpu_set_t CCpuSet;
++#endif
+ #define CpuSet_Zero(p)        CPU_ZERO(p)
+ #define CpuSet_Set(p, cpu)    CPU_SET(cpu, p)
+ #define CpuSet_IsSet(p, cpu)  CPU_ISSET(cpu, p)
+
+--- a/extern/lzma/Threads.c
++++ b/extern/lzma/Threads.c
+@@ -265,7 +265,7 @@
+       */
+
+       // ret2 =
+-      pthread_attr_setaffinity_np(&attr, sizeof(*cpuSet), cpuSet);
```

Comment:
pretty sure this is completely broken

```c
int pthread_setaffinity_np(pthread_t thread, size_t cpusetsize, const cpu_set_t *cpuset);
```
Takes `pthread_t` as its first argument

```c
int pthread_attr_setaffinity_np(pthread_attr_t *attr, size_t cpusetsize, const cpu_set_t *cpuset);
```

takes `pthread_attr_t*`

I'm very surprised that this works

Reply:
fwiw, apline doesn't have the enum or pthread patches

Reply:
Where'd they come from?

Reply:
https://github.com/void-linux/void-packages/pull/52903#issuecomment-2455212570

Reply:
Did you just make them up?

Please look into a fix. The repos will be staged for several days, so there's time to fix this before a broken update goes live.

Reply:
Never mind, should be fixed in 01a57d87673f7b65047c305078d894a852479d79

Reply:
my mistake, the enum patch is in a 'fix-includes' instead of 'musl-fixes' patch in alpine. they build `-DWITH_LZMA=OFF` for blender

so the only things i added are

`typedef struct CCpuSet { unsigned long __bits[128/sizeof(long)]; } CCpuSet;` is a redef of what the header should include, but doesn't seem to (there should be a proper way to do this? not sure why it wasn't working)

as for what cal said about pthread, mailing list mention changing from `pthread_attr_setaffinity_np` to `pthread_setaffinity_np` because of musl not having the 'attr' version. https://lore.kernel.org/all/18966392.1bK43UoomU@xps/T/

sry just left my comment and it never sent

ya i guess lzma will work if that's fine